### DISCUSSION TITLE
Definition of Trigger States

### DISCUSSION THREAD

**Discussion by drusellers** [quartznet/quartznet#2242] [Q&A]
Can someone point me to the code that describes the various values of `trigger_state` from the table `qrtz_triggers`? 

I'd be happy to add the documentation - once I can find it to understand the values.

**Comment by drusellers** [comment]
Like what is `BLOCKED` vs `WAITING`

How do I see what is blocking a trigger?

**Reply by drusellers** [reply]
Blocked - job can't run due to `DisallowConcurrentExecutionAttribute`
Waiting - job will run as soon as a thread opens up

**Comment by lahma** [comment]
Maybe this documentation for enumeration can be of help. These are the logical states, but in database `WAITING` IIRC means "ready to be picked up" and any other state indicates that job cannot be taken just now (can be already acquired, executing, paused...).  

https://github.com/quartznet/quartznet/blob/b3c98be6c6b17eac82a3e29c73cdfa4da4929231/src/Quartz/TriggerState.cs#L28-L79

**Reply by drusellers** [reply]
TY!
